lms/djangoapps/program_enrollments/api/tests/test_linking.py

```diff
@@ -179,6 +185,77 @@ def test_inactive_waiting_course_enrollment(self):
         self.assertEqual(inactive_enrollment.course_enrollment.course.id, self.animal_course)
         self.assertFalse(inactive_enrollment.course_enrollment.is_active)
 
+    def test_realize_course_access_roles(self):
+        program_enrollment = self._create_waiting_enrollment(self.program, '0001')
+        active_enrollment_1 = self._create_waiting_course_enrollment(
+            program_enrollment,
+            self.fruit_course,
+            status='active'
+        )
+        active_enrollment_2 = self._create_waiting_course_enrollment(
+            program_enrollment,
+            self.animal_course,
+            status='active'
+        )
+        CourseAccessRoleAssignmentFactory(enrollment=active_enrollment_1)
+        CourseAccessRoleAssignmentFactory(enrollment=active_enrollment_2)
+        link_program_enrollments(self.program, {'0001': self.user_1.username})
+
+        # assert that staff CourseAccessRoles are created for the user in the courses
+        fruit_course_staff_role = get_course_access_role(
+            self.user_1,
+            self.fruit_course.org,
+            self.fruit_course,
+            CourseStaffRole.ROLE
+        )
+        assert fruit_course_staff_role is not None
+
+        animal_course_staff_role = get_course_access_role(
+            self.user_1,
+            self.animal_course.org,
+            self.animal_course,
+            CourseStaffRole.ROLE
+        )
+        assert animal_course_staff_role is not None
+
+        # assert that all CourseAccessRoleAssignment objects are deleted
+        assert list(active_enrollment_1.courseaccessroleassignment_set.all()) == []
```

Comment:
#NIT Why not just do
```suggestion
        assert not active_enrollment_1.courseaccessroleassignment_set.all().exists()
```